src/main/java/appeng/container/implementations/ContainerPatternTerm.java

```diff
@@ -174,13 +174,21 @@ public void onChangeInventory( final IInventory inv, final int slot, final InvOp
 
 	}
 
-	public void encodeAndMoveToInventory()
+	public void encodeAndMoveToInventory(boolean encodeWholeStack)
 	{
 		encode();
 		ItemStack output = this.patternSlotOUT.getStack();
 		if ( output != null )
 		{
-			if (!getPlayerInv().addItemStackToInventory( output )){
+			if (encodeWholeStack)
+			{
+				ItemStack blanks = this.patternSlotIN.getStack();
+				this.patternSlotIN.putStack(null);
+				if (blanks != null)
+					output.stackSize += blanks.stackSize;
+			}
+			if (!getPlayerInv().addItemStackToInventory( output ))
+			{
```

Comment:
Should't this feature also be added to processing pattern terminal?

Reply:
probably